math/minuit2/src/NumericalDerivator.cxx

```diff
@@ -0,0 +1,336 @@
+         return fDoubleLimTrafo.Ext2int(val, parameter.UpperLimit(), parameter.LowerLimit(), fPrecision);
+      } else if (parameter.HasUpperLimit() && !parameter.HasLowerLimit()) {
+         return fUpperLimTrafo.Ext2int(val, parameter.UpperLimit(), fPrecision);
+      } else {
+         return fLowerLimTrafo.Ext2int(val, parameter.LowerLimit(), fPrecision);
+      }
+   }
+
+   return val;
+}
+
+double NumericalDerivator::DInt2Ext(const ROOT::Fit::ParameterSettings &parameter, double val) const
+{
+   // return the derivative of the int->ext transformation: dPext(i) / dPint(i)
+   // for the parameter i with value val
+
+   double dd = 1.;
+   if (parameter.IsBound()) {
+      if (parameter.IsDoubleBound()) {
+         dd = fDoubleLimTrafo.DInt2Ext(val, parameter.UpperLimit(), parameter.LowerLimit());
+      } else if (parameter.HasUpperLimit() && !parameter.HasLowerLimit()) {
+         dd = fUpperLimTrafo.DInt2Ext(val, parameter.UpperLimit());
+      } else {
+         dd = fLowerLimTrafo.DInt2Ext(val, parameter.LowerLimit());
+      }
+   }
+
+   return dd;
+}
+
+// MODIFIED:
+// This function was not implemented as in Minuit2. Now it copies the behavior
+// of InitialGradientCalculator. See https://github.com/roofit-dev/root/issues/10
+void NumericalDerivator::SetInitialGradient(const ROOT::Math::IBaseFunctionMultiDim *function,
+                                            const std::vector<ROOT::Fit::ParameterSettings> &parameters,
+                                            std::vector<DerivatorElement> &gradient)
+{
+   // set an initial gradient using some given steps
+   // (used in the first iteration)
+
+   assert(function != nullptr && "function is a nullptr");
+
+   double eps2 = fPrecision.Eps2();
+
+   unsigned ix = 0;
+   for (auto parameter = parameters.begin(); parameter != parameters.end(); ++parameter, ++ix) {
+      // What Minuit2 calls "Error" is stepsize on the ROOT side.
+      double werr = parameter->StepSize();
+
+      // Actually, sav in Minuit2 is the external parameter value, so that is
+      // what we called var before and var is unnecessary here.
+      double sav = parameter->Value();
+
+      // However, we do need var below, so let's calculate it using Ext2int:
+      double var = Ext2int(*parameter, sav);
+
+      if (fAlwaysExactlyMimicMinuit2) {
+         // this transformation can lose a few bits, but Minuit2 does it too
+         sav = Int2ext(*parameter, var);
+      }
+
+      double sav2 = sav + werr;
+
+      if (parameter->HasUpperLimit() && sav2 > parameter->UpperLimit()) {
+         sav2 = parameter->UpperLimit();
+      }
+
+      double var2 = Ext2int(*parameter, sav2);
+      double vplu = var2 - var;
+
+      sav2 = sav - werr;
+
+      if (parameter->HasLowerLimit() && sav2 < parameter->LowerLimit()) {
+         sav2 = parameter->LowerLimit();
+      }
+
+      var2 = Ext2int(*parameter, sav2);
+      double vmin = var2 - var;
+      double gsmin = 8. * eps2 * (fabs(var) + eps2);
```

Comment:
```suggestion
      double gsmin = 8. * eps2 * (std::abs(var) + eps2);
```
We should be consistent with the function for `abs`, and `std::abs` is preferred over `fabs` in C++ as far as I know (there are also other occurences of `fabs` that need to be replaced in this file).

Reply:
`abs`olutely. I'll change them in one go then.